### PR TITLE
tests/raft: add read index request test case

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR adds the test case in https://github.com/coreos/etcd/pull/7574

PTAL @siddontang @BusyJay @zhangjinpeng1987

### CODE REVIEW

File: tests/raft/test_raft.rs [hunk 1]
```diff
@@ -1457,6 +1457,46 @@ fn test_handle_heartbeat_resp() {
...
+    // more info: raft dissertation 6.4, step 2.
+    let m = new_message_with_entries(2,
+                                     1,
+                                     MessageType::MsgReadIndex,
+                                     vec![new_entry(0, 0, Some(ctx))]);
+    sm.step(m).expect("");
+    let msgs = sm.read_messages();
+    assert_eq!(msgs.len(), 1);
+    assert_eq!(msgs[0].get_msg_type(), MessageType::MsgHeartbeat);
+    assert_eq!(msgs[0].get_context(), &vec_ctx[..]);
```
Comment (siddontang) [inline tests/raft/test_raft.rs]: is `&vec_ctx` enough?

Comment (hhkbp2) [inline tests/raft/test_raft.rs]: Tried that but it didn't work out.  
`get_context()` returns `&[u8]`, while `&vec_ctx` is a reference to `Vec`.

Comment (siddontang) [inline tests/raft/test_raft.rs]: may be we can use vec_ctx.as_slice() instead

Comment (hhkbp2) [inline tests/raft/test_raft.rs]: "Equivalent to &s[..]."

Comment (hhkbp2) [inline tests/raft/test_raft.rs]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_slice

**Comment by siddontang** [comment]
PTAL @BusyJay

**Pull request merged**